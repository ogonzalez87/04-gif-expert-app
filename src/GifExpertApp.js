import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./GifGrid";

export const GifExpertApp = ({defaultCategories = []}) => {
  const [categories, setCategories] = useState( defaultCategories);
 

  return (
    <>
      <h2> GifExpertApp</h2>
      <hr />
      <AddCategory setCategories = {setCategories}/>
      <ol>
        {categories.map(category => 
           <GifGrid 
           category = {category}
           key = {category}
            /> 
        )}
      </ol>
    </>
  );
};
