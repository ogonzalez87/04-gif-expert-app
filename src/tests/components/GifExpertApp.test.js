import React from "react"; 
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifExpertApp } from "../../GifExpertApp";

describe("GifExpertApp", () => {
  test("should match snapshoot", () => {
    const wrapper = shallow(<GifExpertApp   />);

    expect(wrapper).toMatchSnapshot();
  });
});

test('should show categories list', () => {
    const categories = [ 'One Pounch', 'Goku'];
    const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length)

})
