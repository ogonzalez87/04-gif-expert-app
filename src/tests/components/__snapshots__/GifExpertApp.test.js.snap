// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should match snapshoot 1`] = `
<Fragment>
  <h2>
     GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`should show categories list 1`] = `
<Fragment>
  <h2>
     GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Pounch"
      key="One Pounch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
